src/biguint.rs

```diff
@@ -434,6 +434,55 @@ impl One for BigUint {
 
 impl Unsigned for BigUint {}
 
+macro_rules! pow_impl {
+    ($T:ty) => {
+        impl<'a> Pow<$T> for &'a BigUint {
```

Comment:
Just for posterity: I was going to ask you to add `for BigUint` (by value) too, but I tried it myself first.  That makes it really annoying when you do have a bigint value, as the `pow` method will then try to move ownership.  You'd have to manually reference `(&i).pow(exp)` to avoid this and get the by-ref method, which is really bad for ergonomics.

So if anyone asks why we don't have that `impl`, here's the reason.